macroid-core/src/main/scala/macroid/MultiEventTweaks.scala

```diff
@@ -0,0 +1,146 @@
+package macroid
+
+import scala.language.dynamics
+import scala.language.experimental.macros
+import android.view.View
+import scala.reflect.macros.{Context ⇒ MacroContext}
+import scala.language.postfixOps
+
+private[macroid] trait MultiEventTweaks {
+
+  import MultiEventTweakMacros._
+
+  type UnitHandler = String
+
+  type FuncHandler = String
+
+  sealed trait HandlerType
+
+  case object HUnit extends HandlerType
+
+  case object HUnitFunc extends HandlerType
+
+  object UnitOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handler: Any): Tweak[W] = macro onUnitImpl[W]
+  }
+
+  object UnitFuncOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handler: Any): Tweak[W] = macro onUnitFuncImpl[W]
+  }
+
+  object MultiOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handlers: (String => Any)*): Tweak[W] = macro onMultipleFuncImpl[W]
+  }
+}
+
+object MultiEventTweaks extends MultiEventTweaks
+
+private object MultiEventTweakMacros {
+
+  import MultiEventTweaks._
+
+  def onUnitImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handler: c.Expr[Any]) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(e: String))) = event
+
+    onMultipleFuncImpl(c)(event)(c.Expr[UnitHandler => Any](q"(${newTermName(s"on${e.capitalize}")}: UnitHandler) => $handler"))
```

Comment:
I think you can make it more readable by assigning the quasiquote to a `val` first. `UnitHandler` has to be `_root_.macroid.MultiEventTweaks.UnitHandler`, otherwise the code will not work if the user has not imported `UnitHandler` — the same goes for every use of `UnitHandler` and `FuncHandler` in the quasiqoutes below.